libs/src/sdk/api/tracks/TracksApi.ts

```diff
@@ -0,0 +1,128 @@
+import { BASE_PATH, RequiredError } from '../generated/default/runtime'
+
+import {
+  Configuration,
+  StreamTrackRequest,
+  TracksApi as GeneratedTracksApi
+} from '../generated/default'
+import type { DiscoveryNodeSelectorService } from '../../services/DiscoveryNodeSelector'
+import type { UploadTrackRequest } from './types'
+import type { StorageService } from '../../services/StorageService'
+import { isFileValid } from '../../utils/file'
+import { TRACK_REQUIRED_PROPERTIES, TRACK_REQUIRED_VALUES } from './constants'
+import { objectMissingProperties } from '../../utils/object'
+
+// Subclass type masking adapted from Damir Arh's method:
+// https://www.damirscorner.com/blog/posts/20190712-ChangeMethodSignatureInTypescriptSubclass.html
+// Get the type of the generated TracksApi excluding streamTrack
+type GeneratedTracksApiWithoutStream = new (config: Configuration) => {
+  [P in Exclude<keyof GeneratedTracksApi, 'streamTrack'>]: GeneratedTracksApi[P]
+}
+
+// Create a new "class" that masks our generated TracksApi with the new type
+const TracksApiWithoutStream: GeneratedTracksApiWithoutStream =
+  GeneratedTracksApi
+
+// Extend that new class
+export class TracksApi extends TracksApiWithoutStream {
+  constructor(
+    configuration: Configuration,
+    private readonly discoveryNodeSelectorService: DiscoveryNodeSelectorService,
+    private readonly storage: StorageService
+  ) {
+    super(configuration)
+  }
+
+  /**
+   * Get the url of the track's streamable mp3 file
+   */
+  async streamTrack(requestParameters: StreamTrackRequest): Promise<string> {
+    if (
+      requestParameters.trackId === null ||
+      requestParameters.trackId === undefined
+    ) {
+      throw new RequiredError(
+        'trackId',
+        'Required parameter requestParameters.trackId was null or undefined when calling getTrack.'
+      )
+    }
+
+    const path = `/tracks/{track_id}/stream`.replace(
+      `{${'track_id'}}`,
+      encodeURIComponent(String(requestParameters.trackId))
+    )
+    const host = await this.discoveryNodeSelectorService.getSelectedEndpoint()
+    return `${host}${BASE_PATH}${path}`
+  }
+
+  /**
+   * Upload a track
+   */
+  async uploadTrack(requestParameters: UploadTrackRequest) {
+    const { trackFile, coverArtFile, metadata, onProgress } = requestParameters
+
+    // Validate inputs
```

Comment:
Wonder if there are any additional validations/formatting in audius-client that we should bring in here?

Reply:
Yes! Let me add those in. I'm gonna do a pass on this validation in general because entity manager already uses default values so most of these checks aren't needed

Reply:
Ok I'm gonna merge this for now and then follow up with a pr to add schema validation to the metadata object, we already have a concept of this in libs but I need to verify that it will work exactly the way we need it to